onnx/defs/schema.h

```diff
@@ -1136,74 +1136,128 @@ class OpSchemaRegistry final : public ISchemaRegistry {
 
   class OpSchemaRegisterOnce final {
    public:
-    OpSchemaRegisterOnce(OpSchema& op_schema, int opset_version_to_load = 0) {
+    OpSchemaRegisterOnce(OpSchema& op_schema, int opset_version_to_load = 0, bool fail_duplicate_schema = true) {
```

Comment:
Not allow duplicate schema was the previous behavior in `OpSchemaRegisterOnce `. I don't know why it was so - I'd appreciate ONNX dev to shed light on the reason here. Therefore, I left `fail_duplicate_schema ` default to `True` for backward-compatibility of this API.

As far as this PR attempts to do: it wants to allow a runtime to register specific versions of opset schema without running into error. Consider `RandomNormal` which only exists for version-1. A runtime may register opset-7, then opset-11 as needed. A runtime would expect the second attempt not to fail just because `RandomNormal-1` already exists. Hence from a runtime perspective, as the caller of `RegisterOnnxOperatorSetSchema`, it's natural to assume not to fail on request to register duplicate schema.

When `fail_duplicate_schema=False`, it does not really register the same schema twice. It just won't throw error, and will not register that schema.

Reply:
So overall, this inconsistency is from the natural expectation of the intended caller, and to maintain backward-compatibility of the existing `OpSchemaRegisterOnce`'s interface.

I can change the runtime-level handle `RegisterOnnxOperatorSetSchema` to default on `fail_duplicate_schema=True` to be consistent. But to me, such a API is more confusing for a runtime to use. This is because, for a runtime who doesn't know of ONNX implementation, it may not expect registering opset-7 then opset-11 to throw duplicate schema error.

Shall we make `OpSchemaRegisterOnce` to allow duplicate schema registration request by default? To me (without ONNX expertise), it seems ok to make it a default behavior to allow such request to return **without throwing an error** - what's important is that we aren't actually loading two copies of the same schema here in either case.

Reply:
What is the preferred defaults here? Would like to hear your thoughts @justinchuby @jcwchen. Ty!

Reply:
IMO it's better to keep backward-compatibility for RegisterOnnxOperatorSetSchema as well with default fail_duplicate_schema = true, which is also consistent with all ONNX APIs to prevent confusions.

Reply:
Understood, will use default `fail_duplicate_schema = true` in both APIs, `OpSchemaRegisterOnce` and `RegisterOnnxOperatorSetSchema`.

Reply:
Done